pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -538,6 +540,33 @@ func NewNetworkPolicyController(antreaClientGetter client.AntreaClientProvider,
 	return c, nil
 }
 
+func (c *Controller) GetFqdnCache(fqdnFilter querier.FQDNCacheFilter) []types.DnsCacheEntry {
+	cacheEntryList := []types.DnsCacheEntry{}
+	var pattern *regexp.Regexp
+	var err error
+	if fqdnFilter != (querier.FQDNCacheFilter{}) {
```

Comment:
This is quite unconventional imo, and I doubt it works the way you think it would. In `newFilterFromURLQuery`, it would return a new FQDNCacheFilter with "DomainName" set to "" if no filter is provided in the url, thus not matching the else branch below. What can be done is making the `fqdnFilter` a pointer, and check if is nil to see if there are any filtering requirements at all